src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpException.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Net.Http
+{
+    internal class WinHttpException : Win32Exception
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int HRESULT_FROM_WIN32(int errorCode)
+        {
+            // Implementation of HRESULT_FROM_WINT32 macro
+            if ((errorCode & 0x80000000) == 0x80000000)
+            {
+                return errorCode;
+            }
+            else
+            {
+                return (errorCode & 0x0000FFFF) | unchecked((int)0x80070000);
+            }
+        }
+    
+        public WinHttpException(int error, string message) : base(error, message)
+        {
+            this.HResult = ConvertErrorCodeToHR(error);
```

Comment:
Since I don't think extension properties are a thing, the `this.` here likely isn't required for compilation. (https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md item 4)